core/src/main/java/org/apache/iceberg/ScanTasksIterable.java

```diff
@@ -120,7 +146,21 @@ private void fetchScanTasks(String planTask) {
               headers,
               ErrorHandlers.defaultErrorHandler());
       fileScanTasks.addAll(response.fileScanTasks());
-      planTasks.addAll(response.planTasks());
+      // TODO need to confirm recursive call here
+      if (response.planTasks() != null) {
+        getScanTasksIterable(response.planTasks());
```

Comment:
Seems like we're not using this result anywhere at the moment to populate the top level list? unless i'm missing something

Reply:
Thanks for catching this, will fix this and adding a test to exercise this code path